llvm/lib/Transforms/Vectorize/LoopVectorize.cpp

```diff
@@ -9079,6 +9043,9 @@ void LoopVectorizationPlanner::adjustRecipesForReductions(
         LinkVPBB->insert(FMulRecipe, CurrentLink->getIterator());
         VecOp = FMulRecipe;
       } else {
+        if (PhiR->isInLoop() && isa<VPBlendRecipe>(CurrentLink))
+          continue;
+
```

Comment:
Replace the blend here by its non-header-phi operand (possibly after the loop if needed, and/or updating PreviousLink), as mentioned below?